platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +97,165 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
     @Override
     public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
+
+        // now that the DelegateServletFilter has been initialized with a filterConfig, initialize all SecurityFilters using that filterConfig
+        for (ServiceReference<SecurityFilter> securityFilterServiceReference : sortedSecurityFilterServiceReferences) {
+            initAndAddToFilterList(securityFilterServiceReference, filterConfig);
+        }
+
+        hasBeenInitialized = true;
+    }
+
+    public void addSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference)
+            throws ServletException {
+        if (hasBeenInitialized) {
+            initAndAddToFilterList(securityFilterServiceReference, filterConfig);
+        } else {
+            sortedSecurityFilterServiceReferences.add(securityFilterServiceReference);
```

Comment:
Could move this out of the if/else so you always add to the tree and then only do the init if hasBeenInitialized is true.